src/System.Resources.Extensions/src/System/Resources/Extensions/PreserializedResourceWriter.cs

```diff
@@ -41,9 +79,39 @@ public void AddTypeConverterResource(string name, string typeName, string value)
             if (value == null)
                 throw new ArgumentNullException(nameof(value));
 
-            AddResourceData(name, typeName, new ResourceDataRecord(SerializationFormat.TypeConverterString, value));
+            // determine if the type is a primitive type
+            if (s_primitiveTypes.TryGetValue(typeName, out Type primitiveType))
+            {
+                // directly add strings
+                if (primitiveType == typeof(string))
+                {
+                    AddResource(name, value);
```

Comment:
This is interesting, since I have similar code in the resgen task. But that also has to handle the `ResXFileRef`->text file with specified encoding case, so I don't think I can _just_ rely on this. I don't see a reason to not have it, though.

Reply:
My thought was that you can pretty much plumb the no-mime-type/non-ResXFileRef case straight through to this API.  For mime-types and ResXFileRef you'll need to do some work in translating to different calls.  Hopefully those can largely just pass the string value of the type through.  Let  me know if you have cases where those methods require "primitive type" knowledge.